### PR TITLE
implement concat() on fallible streams as TryStreamExt::try_concat()

### PR DESCRIPTION
**Pull request by barzamin**

### CODE REVIEW

File: futures-util/src/try_stream/mod.rs [hunk 1]
```diff
@@ -438,6 +441,47 @@ pub trait TryStreamExt: TryStream {
...
+    /// Attempt to concatenate all items of a stream into a single
+    /// extendable destination, returning a future representing the end result.
+    ///
+    /// This combinator will extend the first item with the contents of all
+    /// the subsequent successful results of the stream. If the stream is empty,
+    /// the default value will be returned.
+    ///
+    /// Works with all collections that implement the [`Extend`](std::iter::Extend) trait.
+    ///
+    /// This method is similar to [`fold`](super::StreamExt::fold), but will
```
Comment (cramertj) [inline futures-util/src/try_stream/mod.rs]: s/fold/concat

**Comment by cramertj** [comment]
LGTM aside from naming nit.

**Comment by barzamin** [comment]
haha, the copy-and-paste monster strikes again

**Comment by cramertj** [comment]
Thanks for the PR! :)

**Pull request merged**